### ISSUE TITLE
[BUG - Develop] CmdSet.remove() does not take string args

### ISSUE THREAD

**Issue by InspectorCaracal** [evennia/evennia#2736]
#### Describe the bug
Not sure if it's a documentation error or a code error - this claims that you can pass it the key of a command to remove, but causes a traceback.
https://github.com/evennia/evennia/blob/7f4769bd9d3403b8011998b0c903a490caa07be9/evennia/commands/cmdset.py#L566-L568 

#### To Reproduce
1. Attempt to remove a command from any CmdSet with the key, e.g. `self.remove("charcreate")`
2. The code gives a traceback because it cannot handle a string.

#### Expected behavior
Either it should remove the command in the set with that key or the docstring is inaccurate.

#### Develop-branch commit
7f4769bd9

#### Additional context
The traceback:
```
Traceback (most recent call last):
  File ".../evennia/evennia/commands/cmdsethandler.py", line 208, in import_cmdset
    cmdsetclass = cmdsetclass(cmdsetobj)
  File ".../evennia/evennia/commands/cmdset.py", line 222, in __init__
    self.at_cmdset_creation()
  File "./accounts/commands.py", line 27, in at_cmdset_creation
    self.remove("charcreate")
  File ".../evennia/evennia/commands/cmdset.py", line 572, in remove
    if cmd.key.startswith("__"):
AttributeError: 'str' object has no attribute 'key'
```

**Comment by Griatch** [comment]
Looks like a code bug; everyone's been using this with the command object :)

**Comment by kovitikus** [comment]
It's written in this issue as `self.remove()`, but shouldn't it be `self.cmdset.remove()`?

**Comment by InspectorCaracal** [comment]
@kovitikus Not if you're working with a cmdset class, no.

**Comment by ChrisLR** [comment]
@Griatch the cmdset methods all use Cmds directly, I suggest we remove simply remove the docstring mentioning the str to keep things standard, since we're all using cmds anyway

**Comment by Griatch** [comment]
We _could_, but it's indeed meant to work with keys too, and I would think there's code in place to do so ... but something's failing in it. So just removing the doc mention could leave unused or confusing code behind (I have not checked this though)

**Comment by InspectorCaracal** [comment]
Based on what I found when I first poked at it, it's not so much that it doesn't work as it was never implemented.

The first line of `cmdset.remove` is `cmd = self._instantiate(cmd)` - which is supposed to convert `cmd` into a command instance.

But `cmdset._instantiate` passes strings through, as per its _own_ documentation:

https://github.com/evennia/evennia/blob/7f4769bd9d3403b8011998b0c903a490caa07be9/evennia/commands/cmdset.py#L317-L333

**Comment by Griatch** [comment]
Hm. I think the docstring is correct; it's consistent with the rest of the system to support removal both by cmd, key or even cmd-class-path. I don't recall why this was never implemented, but most likely it's because I mainly thought of `CmdsetHandler.remove` as being the main thing people would use. As it turns out, `Cmdset.remove()` is a much more common use case than expected, since this is the easiest way to remove default commands.

**Comment by Griatch** [comment]
Resolved in #2755
